datalad/support/path.py

```diff
@@ -207,6 +208,58 @@ def get_parent_paths(paths, parents, only_with_parents=False, *, sep='/'):
     return res
 
 
+def get_limited_paths(paths: list[str|Path], limits: list[str|Path], *, include_within_path: bool = False) -> list[Path]:
```

Comment:
better name anyone? should I add trailing `_` to signal that it is a generator? (should adjust type annotation for that I guess)

Reply:
may be
- `get_paths_subset`
- `subselect_paths`
- `get_selected_paths`
- `get_paths_under`
- `select_subpaths`
? (I like `get_` fashion though, although it is also a generator and value of being a generator is minimal here really)

Reply:
Not much creative genius from my part:
``get_overlapping_paths``
``get_paths_contained_in``
``get_paths_within_path``

Reply:
I like to have consistent `get_` since we are to return adjusted result instead of filtering in place... then `get_filtered_paths_`, but how to call 2nd paths argument -- I would not like `filters` , so `filter_paths`?

Reply:
> then `get_filtered_paths_` [...] so `filter_paths`?

Yes. I like that better!